### PR TITLE
Related functions about generate view need reviews.

### PR DESCRIPTION
**Pull request by JeanneXu1122**

### CODE REVIEW

File: src/generate_view.c [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    if (p_map->barrier != NULL)
+    {
+        update_barrier(p_map);
+    }
+    if (p_map->arrow != NULL)
+    {
+        update_arrow(p_map);
+    }
+
+    //updatePlayerPos(&map.player, arrowKey);
```
[the remark this answers is not in the document]
Comment (JeanneXu1122) [reply src/generate_view.c:26]: accept

File: src/generate_view.c [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/**
+* @file update_view.c
+* @author Jin
+*/
+
+#include"map.h"
```
[the remark this answers is not in the document]
Comment (JeanneXu1122) [reply src/generate_view.c:6]: accept

File: src/generate_view.c [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        map_barrier_t* p_bar = &p_map->barrier[i];
+        SDL_Rect barrier_rect = { p_bar->current_pos.x, p_bar->current_pos.y,  250, 10 };
+        SDL_RenderCopy(p_renderer, p_barrier_texture, NULL, &barrier_rect);
+    }
+
+
+    for (int i = 0;i < p_map->number_of_arrows;i++)
+    {
+        map_barrier_t* p_arr = &p_map->arrow[i];
```
[the remark this answers is not in the document]
Comment (JeanneXu1122) [reply src/generate_view.c:38]: accept

File: src/generate_view.c [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        SDL_Rect arrow_rect = { p_arr->current_pos.x, p_arr->current_pos.y,  11, 50 };
```
[the remark this answers is not in the document]
Comment (JeanneXu1122) [reply src/generate_view.c:39]: Reject. The logic is right here.

File: src/generate_view.c [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        if (p_arr->current_dir == DIRECTION_DOWN)
```
[the remark this answers is not in the document]
Comment (JeanneXu1122) [reply src/generate_view.c:40]: accept

File: test/generate_view_test.c [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+* @author Jin
+*/
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <SDL.h>
+#include <sdl_ttf.h>
+#include <SDL_image.h>
+#include <stdbool.h>
+#include "map.h"
```
[the remark this answers is not in the document]
Comment (JeanneXu1122) [reply test/generate_view_test.c:12]: accept

File: test/generate_view_test.c [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+
+    SDL_Texture* p_result = SDL_CreateTextureFromSurface(g_renderer, p_picture);
+    SDL_FreeSurface(p_picture);
+
+    return p_result;
+}
+
+int generate_view_test()
+{
+    //update_barrier_test();
```
[the remark this answers is not in the document]
Comment (JeanneXu1122) [reply test/generate_view_test.c:48]: accept

File: test/generate_view_test.c [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+
+    /**Create the window*/
+    g_window = SDL_CreateWindow("GAME WINDOW", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, SCREEN_WIDTH, SCREEN_HEIGHT, SDL_WINDOW_SHOWN);
+    if (NULL == g_window)
+    {
+        SDL_Quit();
+        return -1;
+    }
+
+    //��ʼ�������
```
[the remark this answers is not in the document]
Comment (JeanneXu1122) [reply test/generate_view_test.c]: accept

File: test/generate_view_test.c [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+    if (TTF_Init() == -1)
+        return -1;
+    //������
```
[the remark this answers is not in the document]
Comment (JeanneXu1122) [reply test/generate_view_test.c]: accept

File: test/generate_view_test.c [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+    /**Load the image resource*/
+    g_player_texture = Load_image("data/player.jpg");
+    g_barrier_texture = Load_image("data/barrier.jpg");
+    g_arrow_down_texture = Load_image("data/arrow_down.jpg");
+    g_arrow_up_texture = Load_image("data/arrow_up.jpg");
+    g_goal_texture = Load_image("data/goal.jpg");
+
+
+    /**Load the map*/
+    char file_path[1000] = "data/map.txt";
```
[the remark this answers is not in the document]
Comment (JeanneXu1122) [reply test/generate_view_test.c:92]: accept

File: test/generate_view_test.c [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+        /**Handle events on queue*/
+        while (SDL_PollEvent(&event) != 0)
+        {
+            /**User requests quit*/
+            switch (event.type)
+            {
+            case SDL_QUIT:
+                game_running = true;
+                break;
+            case SDL_KEYDOWN:/**< User presses a key*/
```
[the remark this answers is not in the document]
Comment (JeanneXu1122) [reply test/generate_view_test.c:118]: accept

File: test/generate_view_test.c [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+        SDL_RenderClear(g_renderer);
+        SDL_SetRenderDrawColor(g_renderer, 0xff, 0xff, 0xff, 0xff);
+
+        /**Check if collision between player and arrows*/
+
+        /**Draw all objects*/
+        generate_view(&map, g_renderer, g_player_texture, g_goal_texture, g_barrier_texture, g_arrow_down_texture, g_arrow_up_texture);
+
+        /**Display the content to render*/
+        SDL_RenderPresent(g_renderer);
```
[the remark this answers is not in the document]
Comment (JeanneXu1122) [reply test/generate_view_test.c:188]: accept

File: test/generate_view_test.c [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+    if (SDL_Init(SDL_INIT_EVERYTHING) < 0)
+    {
+        exit(-1);
+    }
+
+    /**Initialize Image library*/
+    IMG_Init(IMG_INIT_PNG | IMG_INIT_JPG);
+
+    /**Create the window*/
+    g_window = SDL_CreateWindow("GAME WINDOW", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, SCREEN_WIDTH, SCREEN_HEIGHT, SDL_WINDOW_SHOWN);
```
[the remark this answers is not in the document]
Comment (JeanneXu1122) [reply test/generate_view_test.c:61]: updated

File: test/generate_view_test.c [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+    /**While application is running*/
+    while (game_running)
+    {
+        /**Handle events on queue*/
+        while (SDL_PollEvent(&event) != 0)
+        {
+            /**User requests quit*/
+            switch (event.type)
+            {
+            case SDL_QUIT:
```
[the remark this answers is not in the document]
Comment (JeanneXu1122) [reply test/generate_view_test.c:115]: It works now.

**Review by Pari-Firoozi — APPROVED** [review]

**Review by Jaserelhabrouk — APPROVED** [review]
Looks good to me

**Review by layeghimahsa1996 — APPROVED** [review]
Great Job!

**Pull request merged**